src/track/macroaction.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "audio/frame.h"
+#include "engine/engine.h"
+#include "proto/macro.pb.h"
+namespace proto = mixxx::track::io;
+
+/// A MacroAction is the building block of a Macro.
+/// It contains a position as well as the action to be taken at that position.
+///
+/// Note that currently only jumps to a target position are available,
+/// but that is subject to change.
+class MacroAction {
+  public:
+    enum class Type : uint32_t {
+        Jump = 0
+    };
+
+    MacroAction(mixxx::audio::FramePos source, mixxx::audio::FramePos target)
+            : source(source), target(target), type(Type::Jump){};
```

Comment:
rendundant ;

Reply:
Without that semicolon `clang-format` creates a linebreak between the curly braces, do you prefer that?

Reply:
Yes. Let clang-format do it's job according to the config. Never tweak the code just for formatting purposes!

Reply:
The only exception are extra line breaks for readability that are preserved during formatting.